### ISSUE TITLE
How to debug blank screen issue with no error output?

### ISSUE THREAD

**Issue by ospfranco** [callstack/repack#621]
Hi, I'm starting to try repack and the documentation is scarce when it comes to per app issues. I just tried starting my app and after "fixing" some typescript transpilation issue with some of the dependencies I'm now facing a blank screen with no apparent cause.

I'm not exactly looking for a solution (though I would appreciate it if anybody has one) but rather, how would one debug such an issue:

![CleanShot 2024-05-14 at 11 38 47@2x](https://github.com/callstack/repack/assets/1634213/7fa32ea6-7ae2-4b9d-a352-559844975655)

**Comment by ospfranco** [comment]
I don't even have react-native-web in my dependencies, I wonder how it got there. In any case, without any info it's hard to know which steps should I take to solve it.

**Comment by jbroma** [comment]
I think it some kind of import inside of reanimated pointing to it and that how it gets involved into all of this.

I agree it's hard, I'm in the process of writing a guide on how to deal with module resolution issues and this is a nice example to showcase indirect cases of module resolution issues like this one.

**Comment by jbroma** [comment]
After a closer look, the warnings can be safely ignored there (the `react-native-web` imports in `node_modules/react-native-reanimated/src/reanimated2/js-reanimated/index.ts` are wrapped with try catch blocks). 

Can you share your repro so we can get to the bottom of this?  
I was able to run the new CLI app with Re.Pack and reanimated without any issues

**Comment by jbroma** [comment]
I can reproduce the same behaviour in metro:

![image](https://github.com/callstack/repack/assets/20626315/8f2446bd-9dc1-4cea-80a4-32a370f507a9)

This is done through commenting out `AppRegistry.registerComponent(appName, () => App);` to reproduce the invariant from the logs. 

Disabling `bridgeless` in `AppDelegate.mm` results in error being displayed:

![image](https://github.com/callstack/repack/assets/20626315/33c0b9b2-f03a-4832-b008-da7058599727)

cc @ospfranco

**Comment by ospfranco** [comment]
Not using bridgeless yet, but using the new arch as I have some custom turbo modules. Let me try to isolate the issue.

**Comment by ospfranco** [comment]
I did not notice bridgeless was turned on. However, after turning it off I still do not get any meaningful error:

![CleanShot 2024-05-14 at 13 47 36@2x](https://github.com/callstack/repack/assets/1634213/7c82d2b4-236d-4a38-8206-bc3307c379d1)

**Comment by jbroma** [comment]
Is it any better when using metro by any chance?

**Comment by ospfranco** [comment]
Yeah metro builds just fine even with bridgeless

![CleanShot 2024-05-14 at 13 54 18@2x](https://github.com/callstack/repack/assets/1634213/38ee658a-8102-4ea1-94bd-0e7aa9583c06)

**Comment by jbroma** [comment]
Looks like some kind of edge-case incompatibility with core, any chance you could create a minimal repro?

**Comment by ospfranco** [comment]
I did add a internal turbo module following [this guide](https://github.com/reactwg/react-native-new-architecture/blob/main/docs/turbo-modules-xplat.md) which might be outdated but did not find any more info.

I'm trying to create a reproducible example.

**Comment by jbroma** [comment]
Thanks, I'll look into it. In the mean time I've opened an issue in core: 
https://github.com/facebook/react-native/issues/44572

**Comment by jbroma** [comment]
@ospfranco 

I've tried running the app but couldn't reproduce it - it runs fine after taking into account the following:
- I've installed pods using this method:
    ```bash
    bundle install
    cd ios
    RCT_NEW_ARCH_ENABLED=1 bundle exec pod install
    ```
    Please note that this resulted in using `CocoaPods `1.14.3` instead of `1.15.2` as it was in your case
- needed to comment out `NativeSampleModule.h` in `AppDelegate.mm` since it wasn't present and the ios compilation failed becaused of that

After that, the app works with everything rendering properly (even in bridgeless)

**Comment by ospfranco** [comment]
Interesting, maybe i built the wrong project. Which bife version are u using?

**Comment by jbroma** [comment]
what do you mean by bife?

I'm on Sonoma 14.5 / Xcode 15.3 / Ruby 2.7.6 / Node 20.12.2

**Comment by ospfranco** [comment]
Sorry, meant node. I was on the phone. I cannot reproduce it on a new project, even when I copied all the same dependencies, I will have to take a deeper look and copy file by file. I will let you know if I manage to isolate the issue.

**Comment by ospfranco** [comment]
Ok, finally managed to reproduce the issue in the example:

<img source="https://github.com/callstack/repack/assets/1634213/2c602665-3aa3-4295-a0cb-3b5a431307fc" width="300"/>

**Comment by ospfranco** [comment]
It seems I'm still getting an error, even after adding the @sentry package

<img src="https://github.com/callstack/repack/assets/1634213/3fb196e0-7ee3-4c99-8dfc-4c24868aadde" width="300" />

**Comment by jbroma** [comment]
The only other difference is the `Gemfile.lock` and `Podfile.lock` (used CocoaPods 1.14.3 vs 1.15.2 as before).  Perhaps you

Generally it's hard to catch errors like this early, it's usually errors likes variable being undefined, or a `class` keyword being present in one of the files and causing an error, or worse - failing silently. In next major we plan to transpile every `node_module` so this shouldn't be an issue anymore.

**Comment by jbroma** [comment]
You can also try transpiling every `node_module` by changing the module rules to `include: /node_modules/`, to see if that solves the issue.

**Comment by ospfranco** [comment]
Tried both cocoapods 1.14.3 and `include /node_modules/` none seem to work. I guess I have to keep spelunking. Thanks a lot for the help so far!

**Comment by jbroma** [comment]
Sure thing! If you manage to find other cause for this, then we can resume finding the root cause

Can you please check if this fixed the issue in the repro for you? From the error above I can see that it's the other app with the error.

**Comment by ReneLa** [comment]
Has anyone been able to resolve this issue?. White blank screen even on fresh app init~

**Comment by maxinminax** [comment]
I have a same issue, it's too difficult to find the reason when app have the blank screen without any error logs

**Comment by jeroen-van-dijk** [comment]
Same issue. White blank screen, no errors on freshly created project

**Comment by jbroma** [comment]
should be fixed in `5.0.0-rc.4`

**Comment by minhduhung2002** [comment]
faced this after adjusting my metro.config.js 
config.resolver.extraNodeModules = {
  zlib: require.resolve("react-zlib-js"),
  crypto: require.resolve("react-native-quick-crypto"),
  http: require.resolve("@iwater/react-native-http-server"),
  https: require.resolve("react-native-axios"),
};
since I upgraded rn version from .76 to .79, which caused
The package at "node_modules/jose/dist/node/esm/runtime/zlib.js" attempted to import the Node standard library module "zlib".
